cup_of_join.py

```diff
@@ -0,0 +1,30 @@
+def join(*lists, sep="-"):
+    # Check if no arguments were passed
+    if lists == ():
+        return None
+
+    # Initialize an empty list
+    new_list = []
+
+    # Loop through each list passed as an argument
+    for list1 in lists:
+        # Loop through each item in the current list
+        for organ in list1:
+            # If the item is another list, recursively call the join function
+            # with the nested list and separator as arguments and append the result to new_list
+            if type(organ) == list:
+                new_list.append(join(organ, sep))
+            # If the item is not a list, directly append it to new_list
+            else:
+                new_list.append(organ)
```

Comment:
1. Python is snake case (list1)-
https://www.google.com/amp/s/www.theserverside.com/answer/Camel-case-vs-snake-case-Whats-the-difference%3famp=1

2. Replace for loops -  
https://medium.com/python-pandemonium/never-write-for-loops-again-91a5a4c84baf
Replace for loops with more advanced python features

Reply:
I fixed it